sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -0,0 +1,133 @@
+   */
+  async processStudentLoanBalances(): Promise<ProcessSFTPResponseResult[]> {
+    const remoteFilePaths =
+      await this.studentLoanBalancesIntegrationService.getResponseFilesFullPath(
+        this.esdcConfig.ftpResponseFolder,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.PT.MBAL.[0-9]{8}.*`,
+          "i",
+        ),
+      );
+    // Sort remote file paths by name.
+    remoteFilePaths.sort((a, b) => a.localeCompare(b));
+    const processFiles: ProcessSFTPResponseResult[] = [];
+    for (const remoteFilePath of remoteFilePaths) {
+      processFiles.push(await this.processFile(remoteFilePath));
+    }
+    return processFiles;
+  }
+
+  /**
+   * Process each individual Student Loan Balances response file from the SFTP.
+   * @param remoteFilePath Student Loan Balances response file to be processed.
+   * @returns process summary and errors summary.
+   */
+  private async processFile(
+    remoteFilePath: string,
+  ): Promise<ProcessSFTPResponseResult> {
+    const result = new ProcessSFTPResponseResult();
+    result.processSummary.push(`Processing file ${remoteFilePath}.`);
+    let studentLoanBalancesSFTPResponseFile: StudentLoanBalancesSFTPResponseFile;
+    try {
+      studentLoanBalancesSFTPResponseFile =
+        await this.studentLoanBalancesIntegrationService.downloadResponseFile(
+          remoteFilePath,
+        );
+    } catch (error) {
+      this.logger.error(error);
+      result.errorsSummary.push(
+        `Error downloading file ${remoteFilePath}. Error: ${error}`,
+      );
+      // Abort the process nicely not throwing an exception and
+      // allowing other response files to be processed.
+      return result;
+    }
+    result.processSummary.push(
+      `File contains ${studentLoanBalancesSFTPResponseFile.records.length} Student Loan balances.`,
+    );
+    // Get only the file name for logging.
+    const fileName = path.basename(remoteFilePath);
+    for (const studentLoanBalanceRecord of studentLoanBalancesSFTPResponseFile.records) {
+      try {
+        const student: Student =
+          await this.studentService.getStudentByPersonalInfo(
+            studentLoanBalanceRecord.sin,
+            studentLoanBalanceRecord.lastName,
+            getISODateOnlyString(studentLoanBalanceRecord.birthDate),
+          );
+        // If student not found continue.
+        if (!student) {
+          result.processSummary.push(
+            `Student not found for line ${studentLoanBalanceRecord.lineNumber}`,
+          );
+          continue;
+        }
+        const studentLoanBalances = new StudentLoanBalances();
+        studentLoanBalances.student = student;
+        studentLoanBalances.cslBalance = studentLoanBalanceRecord.cslBalance;
+        studentLoanBalances.balanceDate =
+          studentLoanBalancesSFTPResponseFile.header.balanceDate;
+        // Save the student loan balance.
+        await this.studentLoanBalancesService.save(studentLoanBalances);
```

Comment:
Please use an insert to boost the performance since a great amount of records are expected.